doc/fips/SAI-Proposal-FIPS-Compliance.md

```diff
@@ -0,0 +1,397 @@
+
+```
+/**
+ * @brief Attribute data for #SAI_IPSEC_ATTR_POST_STATUS,
+ */
+typedef enum _sai_ipsec_post_status_t
+{
+    /** Unknown */
+    SAI_IPSEC_POST_STATUS_UNKNOWN,
+
+    /** Pass */
+    SAI_IPSEC_POST_STATUS_PASS,
+
+    /** In Progress */
+    SAI_IPSEC_POST_STATUS_IN_PROGRESS,
+
+    /** Fail */
+    SAI_IPSEC_POST_STATUS_FAIL,
+} sai_ipsec_post_status_t;
+
+/**
+ * @brief Attribute Id for sai_ipsec
+ */
+typedef enum _sai_ipsec_attr_t
+{
+    . . .
+
+    /**
+     * @brief IPSEC POST status
+     * Attribute to query the status of POST for an IPSEC engine
+     *
+     * @type sai_ipsec_post_status_t
+     * @flags READ_ONLY
+     */
+    SAI_IPSEC_ATTR_POST_STATUS,
+
+    /**
+     * @brief Setting the value to true will start the post on all the ports serviced by this IPSEC engine
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_IPSEC_ATTR_ENABLE_POST,
+
+
+```
+
+#### 2.2.1 IPSec Port
+
+New READ only attribute SAI_IPSEC_PORT_ATTR_POST_STATUS is introduced to read the status of POST for a given IPSec port.
+This attribute can be read after the POST is completed for the IPSec object hosting this port.
+
+```
+/**
+ * @brief Attribute data for #SAI_IPSEC_PORT_ATTR_POST_STATUS
+ */
+typedef enum _sai_ipsec_port_post_status_t
+{
+    /** Unknown */
+    SAI_IPSEC_PORT_POST_STATUS_UNKNOWN,
+
+    /** Pass */
+    SAI_IPSEC_PORT_POST_STATUS_PASS,
+
+    /** Fail */
+    SAI_IPSEC_PORT_POST_STATUS_FAIL,
+
+} sai_ipsec_port_post_status_t;
+
+/**
+ * @brief Attribute Id for sai_ipsec_port
+ */
+typedef enum _sai_ipsec_port_attr_t
+{ 
+    . . .
+
+    /**
+     * @brief IPSEC Port POST completion status
+     *
+     * Attribute to query the status of POST for a IPSEC port
+     *
+     * @type sai_ipsec_port_post_status_t
+     * @flags READ_ONLY
+     */
+    SAI_IPSEC_PORT_ATTR_POST_STATUS,
+    . . .
+
+} sai_ipsec_port_attr_t;
+
+```
+
+
+### 3.0 Completion Callback
+
+Single aggregate callback function is provided to return the status of POST status for the entire MACSec or IPSec engine. 
+
+If the engine is servicing a single port then this callback essentially becomes a per port callback.
+
+If the engine is servicing multiple ports then each port needs to be queried for its POST status using the READ only attribute of the port.
+
+#### 3.1 MACSec POST Completion Callback
+
+```
+/**
+ * @brief MACSEC post status notification
+ *
+ * @objects switch_id SAI_OBJECT_TYPE_MACSEC
+ *
+ * @param[in] macsec_id MACSEC Id
+ * @param[in] macsec_post_status MACSEC post status
+ */
+typedef void (*sai_macsec_post_status_notification_fn)(
+        _In_ sai_object_id_t macsec_id,
+        _In_ sai_macsec_post_status_t macsec_post_status);
+
+/**
+ * @brief Attribute Id in sai_set_switch_attribute() and
+ * sai_get_switch_attribute() calls.
+ */
+typedef enum _sai_switch_attr_t
+{
+    /**
+     * @brief Callback for completion status of all the MACSEC ports serviced by this MACSEC engine
```

Comment:
does the notification callback gets registered at the sai switch level or at the macsec engine level